backend/src/main/java/com/github/NicolaiKopka/api_services/MovieDBApiConnect.java

```diff
@@ -31,4 +33,13 @@ public List<Movie> getPopularMovies() {
         List<Movie> movies = response.getResults().stream().toList();
         return movies;
     }
+
+    public Collection<Movie> getMoviesByQuery(String query) {
+        String discoverByPopularityEndpoint = urlPrefix + "search/movie?api_key=" +
+                API_KEY + "&query=" + query + "&include_adult=false";
+
+        MovieList response = restTemplate.getForObject(discoverByPopularityEndpoint, MovieList.class);
+        List<Movie> movies = response.getResults().stream().toList();
```

Comment:
The variable `movies` is not needed. `response.getResults().stream().toList();` can be returned directly.